node/src/main/scala/org/bitcoins/node/models/BroadcastAbleTransactionTable.scala

```diff
@@ -0,0 +1,43 @@
+package org.bitcoins.node.models
+
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.ProvenShape
+import org.bitcoins.db.DbRowAutoInc
+import org.bitcoins.db.TableAutoInc
+import org.bitcoins.core.crypto.DoubleSha256Digest
+import scodec.bits.ByteVector
+import org.bitcoins.core.protocol.transaction.Transaction
+
+/** TXs we can broadcast over the P2P network */
+final case class BroadcastAbleTransaction(
```

Comment:
I'm not sure if this abstraction is right, perhaps it would be better to called this `SignedTransaction` and then have a field in this that indicates if `isBroadcasted: Boolean`, and then maybe a `isConfirmed` or something like that. Although this starts to get redundant with confirmations else where in our data model. Is the idea here we delete txs from this table after they have been broadcasted?

Reply:
Thinking about this further, there seems to be another state, something like "sent inv", which does not necessarily mean it is broadcasted. This is also peer specific which makes it more complex. Maybe this is making things too complex for this PR, and we should address separately.

Reply:
My reasoning here is that we might not broadcast a TX only once. Our peer (or peers, once that's implemented) might ask for the same TX again. Would be bad if we couldn't send them to them?

Reply:
I don't think that is the expected behavior, and we should probably consider the peer at best as broken, at worst as malicious if they request the transaction multiple times. Here is the algorithm bitcoin core uses for _requesting_ information about a inv tx they received: https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L295

Here is how bitcoin core processes an _incoming_ TX request on the p2p network. I can't tell what there behavior is from a cursory glance, but we definitely want to at least look at this https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L2470